databuilder/__main__.py

```diff
@@ -62,6 +62,9 @@ def show_help(**kwargs):
     add_generate_measures(subparsers, environ)
     add_test_connection(subparsers, environ)
 
+    if "--" in args:
```

Comment:
If `--` is not in args then `user_args` will not be changed from `None` which means that `parser.parse_args(user_args)` will end up parsing the whole of `sys.argv[1:]` which I don't think is what we want.